libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1473,6 +1473,29 @@ bool ExpressionCompiler::visit(MemberAccess const& _memberAccess)
 							return false;
 						}
 
+	// Another special case for `address.code.length`, which should simply call extcodesize
+	if (
+		auto innerExpression = dynamic_cast<MemberAccess const*>(&_memberAccess.expression());
+		member == "length" &&
+		innerExpression &&
+		innerExpression->memberName() == "code" &&
+		innerExpression->expression().annotation().type->category() == Type::Category::Address
+	)
+	{
+		solAssert(_memberAccess.expression().annotation().type->category() == Type::Category::Array, "");
+
+		innerExpression->expression().accept(*this);
+
+		utils().convertType(
+			*innerExpression->expression().annotation().type,
+			*TypeProvider::address(),
+			true
+		);
+		m_context << Instruction::EXTCODESIZE;
```

Comment:
Maybe it makes sense to also add this functionality to `libsolidity/codegen/ir/IRGeneratorForStatements.cpp`.

Reply:
Just saw that the test is already `compileViaYul: also`. I thought there would have been also a change in `libsolidity/codegen/ir/IRGeneratorForStatements.cpp` needed to add this functionality. Looks like that I was wrong.